getoptpp.hpp

```diff
@@ -45,15 +49,8 @@ void Options::add(std::string const & name, int has_arg, int val, Handle handle)
 }
 
 void Options::add(std::string const & name, int has_arg, int* flag, int val, Handle handle) {
-	// create a new option from the args
-	options.emplace_back();
-	auto& option = options.back();
-	option.name = name.c_str();
-	option.has_arg = has_arg;
-	option.flag = flag;
-	option.val = val;
-	
 	int index = val; // let val be the option's unique identifier
+	options.push_back({name.data(), has_arg, flag, val});
```

Comment:
please keep the comment (// create a new option from the args) for the push_back.